doc/doxygen/src/build-system-cpu-conventions.md

```diff
@@ -0,0 +1,177 @@
+and behaviors. This leads to the difference between `CPU` and `CPU_MODEL`.
+
+Several handling were implemented in the build system using shell invocations
+and algorithmic resolution based on the models characters.
+However, due to the algorithmic resolution one cannot know anymore the value of
+the specific CPU model definitions without executing the build system and the
+shell `sed` magic.
+
+This cause both issue to read the definition and slow execution time for
+definitions that are a static mapping from the exact CPU model.
+
+With the way of defining boards and CPU, some informations should be defined
+either for the description, dependency resolution or compilation
+(respectively in the `Makefile.features`, the `Makefile.dep` or `Makefile.include`)
+so may want to define only each piece of information required for each step.
+
+
+Different solutions
+-------------------
+
+This describes different possible solutions and in which case which applies.
+
+If only one should be used, then only the first one would currently work
+due to the fact that dependencies are not defined declaratively.
+But due to the overhead, intermediate single file declarative solution are
+good candidates.
+
+The main goal is to remove the algorithmic and shell based resolutions for this
+if possible.
+
+### File based handling ###
+
+Somehow, having all `CPU_MODEL` in one directory is just to save common
+parts and keep the cpu directory tidy, all the different ones could still
+be defined in there own directory, but nested in the common one.
+
+This could transfer to have for each `CPU_MODEL` for a `CPU` the following files:
+
+~~~~~~~~~~~~~~~~~~~ {.mk}
+$(RIOTCPU)/$(CPU)/models/$(CPU_MODEL)/Makefile.features
+$(RIOTCPU)/$(CPU)/models/$(CPU_MODEL)/Makefile.dep
+$(RIOTCPU)/$(CPU)/models/$(CPU_MODEL)/Makefile.include
+$(RIOTCPU)/$(CPU)/models/$(CPU_MODEL)/doc.txt
+~~~~~~~~~~~~~~~~~~~
+
+And then each main `$(CPU)` file will include the `$(CPU_MODEL)` specific one.
+
+This would match the way to do it in RIOT but could create many
+directories and files for maybe no or only few lines.
+
+It could make sense to go with this if the files, especially the
+`Makefile.dep` or `Makefile.include` ones would need many specific definitions
+and handling that are more than variables configuration for the main `CPU`
+files.
+
+
+### `ifneq` based handling ###
+
+To not add any files, the main `CPU` files could add specific definitions if the
+`CPU_MODEL` matches certain values.
+
+~~~~~~~~~~~~~~~~~~~ {.mk}
+ifneq (,$(filter abc1 abd0 ade5,$(CPU_MODEL)))
+  FEATURES_PROVIDED += periph_hwrng
+endif
+~~~~~~~~~~~~~~~~~~~
+
+This allow simple handling when the benefit of having per model definition
+would currently not overcome the overhead of defining it.
+Also, as currently dependencies are not declarative, it is with the previous
+solution one of the only alternative as the following declarative only would
+not work.
```

Comment:
As currently dependencies are not declarative. Only solutions 1 and 2 would not be able to declare specific `CPU_MODEL` dependencies (as solutions 3 and 4 are declarative only.)

Reply:
Also move this before the solutions descriptions.

Reply:
With the solutions numbering it is way better.

I fixed "only solutions 1 and 2 would ~not~ be able".

I rephrased and moved to the first paragraph. I had to change the other description already talking about the different solutions.
It is in its own squash commit so you can see in details what I changed.